### PR TITLE
docs: Remove references to interactive docs in the open source project

### LINKED ISSUE #2377: No documentation on how to enable interactivity with REST service for OpenAPI spec
**Describe the bug**

I spent some time creating a Helm chart for the Redoc that should allow to deploy it alongside the actual REST service that it is documenting and allow the user to interact with the actual REST API.

But there is no documentation on how to enable interactivity, and it does not seem to work by default, even though my spec contains the base URL that the server is reachable by. I am expecting Redoc to have the same functionality as Swagger UI, where I am able to interact with the REST service... Does this not work?

**Expected behavior**

I expect Redoc to allow me to issue requests to the REST service.
Otherwise, creating a web server for Redoc does not add any value beyond what I get from just generating the single page documentation using `redocly` CLI.

### PR DESCRIPTION
**Pull request by lornajane**
## What/Why/How?

Fixes #2377 - since we still describe the API documentation as interactive although Try It is not included in the open source version. This pull request updates some language to clarify the status in the readme.

## Check yourself

- [x] Code is linted
- [x] Tested
- [x] All new/updated code is covered with tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,7 +73,7 @@ Add your own `spec-url` to the `<redoc>` tag; this attribute can also be a local
 
 Check out the [deployment documentation](./docs/deployment/intro.md) for more options, and detailed documentation for each.
 
-## Redoc vs. Reference
+## Redoc vs. Redocly API Reference
```
Comment (adamaltman) [inline README.md:76]:
```suggestion
## Redoc vs. Redocly
```

File: README.md [hunk 2]
```diff
@@ -34,8 +34,6 @@ enter the URL for your definition and select **TRY IT**.
 
 - Responsive three-panel design with menu/scrolling synchronization
 - Support for OpenAPI 3.1, OpenAPI 3.0, and Swagger 2.0
-- [Multiple deployment options](https://redocly.com/docs/redoc/)
```
Comment (adamaltman) [inline README.md:37]: Why is this removed?

Comment (lornajane) [reply README.md:37]: It's repeated in the next section and I think the in-page examples make more sense in a README context.
